docker-compose.yml

```
@@ -38,6 +38,7 @@ services:
     environment:
       PGDATA: /data
       POSTGRES_DB: db
+      POSTGRES_HOST_AUTH_METHOD: trust
     healthcheck:
       test: echo "\\l" | psql -U postgre
```

To make the tests run locally, I need this change too:
```suggestion
      test: echo "\\l" | psql -U postgres
```

---

it works for me just as it is now ..

---

Could you try this:
1. Start a postgres:12 container:
  `docker run -e PGDATA=/data -e POSTGRES_DB=db -e POSTGRES_HOST_AUTH_METHOD=trust postgres:12`
2. (in another console) connect to the running image and run the `echo|psql` command:
  `docker exec -it $(docker ps | grep postgres:12 | head -n 1 | awk '{ print $1 }') sh -c 'echo "\\l" | psql -U postgre'`

When I do this, I get `psql: error: FATAL:  role "postgre" does not exist`.

When I change to use `psql -U postgres` I get
```
                                 List of databases
   Name    |  Owner   | Encoding |  Collate   |   Ctype    |   Access privileges
-----------+----------+----------+------------+------------+-----------------------
 db        | postgres | UTF8     | en_US.utf8 | en_US.utf8 |
 postgres  | postgres | UTF8     | en_US.utf8 | en_US.utf8 |
 template0 | postgres | UTF8     | en_US.utf8 | en_US.utf8 | =c/postgres          +
           |          |          |            |            | postgres=CTc/postgres
 template1 | postgres | UTF8     | en_US.utf8 | en_US.utf8 | =c/postgres          +
           |          |          |            |            | postgres=CTc/postgres
(4 rows)
```

---

That's right! I get the same behaviour following the procedure you mentioned.
Which made me dig deeper into these health checks. They are actually not being used. So we can completely remove them.

---

Thank you @bquorning ! I fixed this `pg12` health check and we no longer rely on `wait-for-it` to do health check. Thus resolving the conversation.